test/acceptance/tests/terminating-gateway/terminating_gateway_test.go

```diff
@@ -11,61 +12,148 @@ import (
+			t.Log("creating static-server deployment")
+			helpers.Deploy(t, ctx.KubectlOptions(), cfg.NoCleanupOnFailure, "fixtures/static-server.yaml")
+
+			// Once the cluster is up, register the external service, then create the config entry.
+			consulClient := consulCluster.SetupConsulClient(t, c.secure)
+
+			// Register the external service
+			t.Log("registering the external service")
+			_, err := consulClient.Catalog().Register(&api.CatalogRegistration{
+				Node:     "legacy_node",
+				Address:  "static-server",
+				NodeMeta: map[string]string{"external-node": "true", "external-probe": "true"},
+				Service: &api.AgentService{
+					ID:      "static-server",
+					Service: "static-server",
+					Port:    80,
+				},
+			}, &api.WriteOptions{})
+			require.NoError(t, err)
+
+			// If ACLs are enabled we need to update the token of the terminating gateway
+			// with service:write permissions to the static-server service
+			// so that it can can request Connect certificates for it.
+			if c.secure {
+				// Create a write policy for the static-server
+				_, _, err := consulClient.ACL().PolicyCreate(&api.ACLPolicy{
+					Name:  "static-server-write-policy",
+					Rules: staticServerPolicyRules,
+				}, nil)
+				require.NoError(t, err)
+
+				// Get the terminating gateway token
+				tokens, _, err := consulClient.ACL().TokenList(nil)
+				require.NoError(t, err)
+				var termGwTokenID string
+				for _, token := range tokens {
+					if strings.Contains(token.Description, "terminating-gateway-terminating-gateway-token") {
+						termGwTokenID = token.AccessorID
+						break
+					}
+				}
+				termGwToken, _, err := consulClient.ACL().TokenRead(termGwTokenID, nil)
+				require.NoError(t, err)
+
+				// Add policy to the token and update it
+				termGwToken.Policies = append(termGwToken.Policies, &api.ACLTokenPolicyLink{Name: "static-server-write-policy"})
+				_, _, err = consulClient.ACL().TokenUpdate(termGwToken, nil)
+				require.NoError(t, err)
+			}
+
+			// Create the config entry for the terminating gateway
+			t.Log("creating config entry")
+			created, _, err := consulClient.ConfigEntries().Set(&api.TerminatingGatewayConfigEntry{
+				Kind:     api.TerminatingGateway,
+				Name:     "terminating-gateway",
+				Services: []api.LinkedService{{Name: "static-server"}},
+			}, nil)
+			require.NoError(t, err)
+			require.True(t, created, "config entry failed")
+
+			// Deploy the static client
+			t.Log("deploying static client")
+			helpers.Deploy(t, ctx.KubectlOptions(), cfg.NoCleanupOnFailure, "fixtures/static-client.yaml")
+
+			// If ACLs are enabled, test that intentions prevent connections.
+			if c.secure {
+				// With the terminating gateway up, we test that we can make a call to it
+				// via the static-server. It should fail to connect with the
+				// static-server pod because of intentions.
+				t.Log("testing intentions prevent connections through the terminating gateway")
+				helpers.CheckStaticServerConnection(t,
```

Comment:
I know this method isn't a part of this PR, but can we move the expectedResult value to the be the last argument (source, expectedResult,destination) -> (source, destination, expectedResult). IMO it reads a little better as an assertion. I wouldnt mind (expectedResult, source, destination) either.

Reply:
I can't make `expectedResult` last because the last argument of that function is variadic (takes arbitrary number of additional curl args). I can change it to `(expectedResult, source, destination)` though. Do you think that will read better?

Reply:
oh that makes sense. Sorry. Did not look at the actual signature. `(expectedResult, source, destination)` is perfect. It reads like an assertion if that makes sense. Thanks for indulging these requests 😁

Reply:
Updated!